### PR TITLE
feat: improve pip completer

### PR DESCRIPTION
**Pull request by jnoortheen**
## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by jnoortheen** [comment]
@daniel-shimon can you check the failing tests? they are related to the `alias` changes. Seems like the threaded part is not waited.

File: tests/completers/test_pip_completer.py [hunk 1]
```diff
@@ -1,16 +1,22 @@
+import os
```
Comment (gforsyth) [inline tests/completers/test_pip_completer.py:1]: Is this being used?

Comment (gforsyth) [reply tests/completers/test_pip_completer.py:1]: And if it isn't, what happened to the flake8 check?

**Review by gforsyth — COMMENTED** [review]
Thanks for this @jnoortheen -- overall looks good, one nit

File: tests/completers/test_pip_completer.py (hunk 1 above)
Comment (jnoortheen) [reply tests/completers/test_pip_completer.py:1]: Imports inside tests are not properly checked . I will check it.

Comment (jnoortheen) [reply tests/completers/test_pip_completer.py:1]: image

tests are excluded from checks. I will try to enable flake8 for them in another PR. I will remove os import in this.

**Comment by gforsyth** [comment]
Thanks @jnoortheen !

**Pull request merged**